### DISCUSSION TITLE
Is it possible to use pymupdf to convert swf to editable text pdf?

### DISCUSSION THREAD

**Discussion by nissansz** [pymupdf/PyMuPDF#2028] [Q&A] [answered]
Is it possible to use pymupdf to convert swf to editable text pdf?

**Comment by JorjMcKie** [comment]
SWF = ?
Shock Wave Flash?

**Comment by JorjMcKie** [comment]
But anyway the answer is probably NO, because only standard PDFs can be produced with PyMuPDF.

**Comment by nissansz** [comment]
Yes. Currently, I use gfx2gfx to render pdf, which is vector pdf.
Or jpex to convert to text pdf, it is slow, and some files may have problem.

**Reply by JorjMcKie** [reply]
did you try to do `fitz.open("file.swf")` at all?
The fastest way to check support.

**Comment by nissansz** [comment, marked as answer]
_fitz.Document_swiginit(self, _fitz.new_Document(filename, stream, file type, rect, width, height, fontsize))
fitz.fitz.FileDataError: cannot open broken document
[0004.zip](https://github.com/pymupdf/PyMuPDF/files/9951989/0004.zip)

**Reply by JorjMcKie** [reply]
yeah, had tried that myself already. SWF is not supported by MuPDF - and will not be any time soon.

**Comment by nissansz** [comment]
Support soon?

**Reply by JorjMcKie** [reply]
no, on the contrary: never!

**Comment by nissansz** [comment]
understood
